core/services/relay/evm/write_target.go

```diff
@@ -69,14 +88,193 @@ func NewWriteTarget(ctx context.Context, relayer *Relayer, chain legacyevm.Chain
+	}
+
+	registryMetrics, err := registry.NewMetrics()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new registry metrics: %w", err)
+	}
+
+	emitter := writetarget.NewMonitorEmitter(lggr)
+
+	dfProcessor := dfprocessor.NewDataFeedsProcessor(registryMetrics, emitter, false)
+	ccipDfProcessor := dfprocessor.NewDataFeedsProcessor(registryMetrics, emitter, true)
+	porProcessor := porprocessor.NewPORFeedsProcessor(registryMetrics, emitter)
+
+	evmProcessors, err := processor.NewPlatformProcessors(emitter)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create EVM platform processors: %w", err)
+	}
+
+	beholder, err := writetarget.NewMonitor(writetarget.MonitorOpts{
+		Lggr:                      lggr,
+		ProductAgnosticProcessors: evmProcessors,
+		ProductSpecificProcessors: map[string]beholder.ProtoProcessor{
+			"evm-data-feeds":      dfProcessor,
+			"evm-data-feeds-ccip": ccipDfProcessor,
+			"evm-por-feeds":       porProcessor,
+		},
+		Emitter: emitter,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Aptos WT monitor client: %+w", err)
+	}
+
+	opts := writetarget.WriteTargetOpts{
+		ID:     id,
+		Logger: lggr,
+		Config: writetarget.Config{
+			ConfirmerPollPeriod: *config.PollPeriod(),
+			ConfirmerTimeout:    *config.AcceptanceTimeout(),
+		},
+		ChainInfo:            chainInfo,
+		Beholder:             beholder,
+		ChainService:         chain,
+		ConfigValidateFn:     evaluate,
+		NodeAddress:          config.FromAddress().String(),
+		ForwarderAddress:     config.ForwarderAddress().String(),
+		TargetStrategy:       NewEVMTargetStrategy(cr, cw, config.ForwarderAddress().String(), gasLimitDefault, lggr),
+		WriteAcceptanceState: *config.TxAcceptanceState(),
+	}
+
+	return writetarget.NewWriteTarget(opts), nil
+}
+
+type Inputs struct {
+	SignedReport ocr3types.SignedReport
+}
+
+type Request struct {
+	Metadata capabilities.RequestMetadata
+	Config   Config
+	Inputs   Inputs
+}
+
+func getEVMRequest(rawRequest capabilities.CapabilityRequest) (
+	Request, error) {
+	var r Request
+	r.Metadata = rawRequest.Metadata
+
+	if rawRequest.Config == nil {
+		return Request{}, errors.New("missing config field")
+	}
+
+	if err := rawRequest.Config.UnwrapTo(&r.Config); err != nil {
+		return Request{}, err
+	}
+
+	if !common.IsHexAddress(r.Config.Address) {
+		return Request{}, fmt.Errorf("'%v' is not a valid address", r.Config.Address)
+	}
+
+	if rawRequest.Inputs == nil {
+		return Request{}, errors.New("missing inputs field")
+	}
+
+	// required field of target's config in the workflow spec
+	signedReport, ok := rawRequest.Inputs.Underlying[writetarget.KeySignedReport]
+	if !ok {
+		return Request{}, fmt.Errorf("missing required field %s", writetarget.KeySignedReport)
+	}
+
+	if err := signedReport.UnwrapTo(&r.Inputs.SignedReport); err != nil {
+		return Request{}, err
+	}
+	return r, nil
+}
+
+func evaluate(rawRequest capabilities.CapabilityRequest) (receiver string, err error) {
```

Comment:
imo there's nothing EVM specific here and this fn should be extracted and generalized as part of the WT framework